app/src/client/components/buildPage/UserProperty.tsx

```diff
@@ -20,96 +20,127 @@ interface Props {
-                  </div>
-                  {userPropertiesByType.length === 0 && (
-                    <div className='flex flex-col gap-3'>
-                      {/* <h2 className='text-lg font-semibold text-airt-primary'>Available Models</h2> */}
-                      <p className='text-airt-primary mt-1 -mt-3 opacity-50'>{`No ${propertyHeader} found. Please add one.`}</p>
-                    </div>
-                  )}
-                  {userPropertiesByType.length > 0 && (
-                    <div className='flex-col flex w-full'>
-                      <ModelsList
-                        models={userPropertiesByType}
-                        onSelectModel={getSelectedUserProperty}
-                        type_name={propertyName}
-                      />
+export const UserProperty = memo(
+  ({ activeProperty, propertiesSchema, sideNavItemClickCount, setActiveProperty }: Props) => {
+    const propertyHeader = _.find(navLinkItems, ['componentName', activeProperty])?.label;
+    const propertyName = activeProperty === 'llm' ? 'LLM' : capitalizeFirstLetter(activeProperty);
+    const propertySchemasList = filerOutComponentData(propertiesSchema, activeProperty);
+
+    const { parser, pushNewParser, popFromStack, clearStack } = useFormHandler(setActiveProperty);
+
+    const { data: userProperties, refetch: refetchUserProperties, isLoading: isLoading } = useQuery(getModels);
+    const userPropertiesByType = (userProperties && filterPropertiesByType(userProperties, activeProperty)) || [];
+
+    //createNewFormResumeFlow
+    const updateFormStack = async (
+      model: string | null,
+      property: string | null = null,
+      formState: any = null,
+      flow: Flow = Flow.ADD_MODEL
+    ) => {
+      let propertyToShow: string = activeProperty;
+      let replaceLast: boolean = true;
+
+      if (property) {
+        propertyToShow = property;
+        replaceLast = false;
+      }
+
+      const parserParams: PushParser = {
+        propertiesSchema,
+        activeProperty: propertyToShow,
+        activeModel: model,
+        flow: flow,
+        userProperties,
+        replaceLast,
+      };
+
+      if (formState) {
+        parserParams.formState = formState;
+      }
+
+      pushNewParser(parserParams);
+    };
+
+    const addProperty = (e: React.MouseEvent<HTMLButtonElement>) => {
+      e.preventDefault();
+      updateFormStack(propertySchemasList.schemas[0].name);
+    };
+
+    const getSelectedUserProperty = (index: number) => {
+      const selectedProperty = userPropertiesByType[index];
+      pushNewParser({
+        propertiesSchema,
+        activeProperty,
+        activeModel: selectedProperty.model_name,
+        activeModelObj: selectedProperty,
+        flow: Flow.UPDATE_MODEL,
+        userProperties: userProperties,
+      });
+    };
+
+    useEffect(() => {
+      clearStack();
+    }, [sideNavItemClickCount]);
+
+    const flow = parser?.getFlow();
```

Comment:
In which situations parser can be null or undefined, and why is in that case ok that flow is undefined?